framework/PluginsBaseClasses/PostProcessorPluginBase.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on March 10, 2021
+
+@author: wangc
+"""
+
+#External Modules---------------------------------------------------------------
+import abc
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from .PluginBase import PluginBase
+from Models.PostProcessors.PostProcessor import PostProcessor
+#Internal Modules End-----------------------------------------------------------
+
+class PostProcessorPluginBase(PostProcessor, PluginBase):
+  """
+    This class represents a specialized class from which each PostProcessor plugins must inherit from
+  """
+  # List containing the methods that need to be checked in order to assess the
+  # validity of a certain plugin. This list needs to be populated by the derived class
+  _methodsToCheck = ['getInputSpecification', 'handlePluginInput', 'initializePlugin', 'runPluginDataProcessor']
+  entityType = 'PostProcessor'
+
+  ##################################################
+  # Methods to link internal PostProcessor methods
+  ##################################################
+  def __init__(self, runInfoDict):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    PluginBase.__init__(self)
+    PostProcessor.__init__(self, runInfoDict)
```

Comment:
@joshua-cogliati-inl correct me here, but I think `super(PostProcessorPluginBase, self).__init__` might be preferable to check and handle inheritance, unless we specifically care about forcing the order in which the inheritance is called.

Reply:
I like super() method, however, in our code, we do not use super() in most of our classes. This will create issues if we just limit the super() in PostProcessorPluginBase class. I suggest to have another discussion to see if want to use super across RAVEN entities. If so, I think an issue need to be opened, and a separate PR should be created instead to change it directly in the PostProcessorPlugin API. @PaulTalbot-INL @joshua-cogliati-inl